File: 13. String length via pointers.cpp
// Topic: Find length of a string using pointer arithmetic (no strlen)
// Description: Walk pointer until '\0'
#include <iostream>
using namespace std;

int main()
{
    const char* s = "hello world";
    const char* p = s;
    while (*p) ++p;
    cout << "Length: " << (p - s) << '\n';
    return 0;
}

File: 13. String length via pointers.py
"""
Topic: Find string length using pointer arithmetic (no len)
Description: Iterate until StopIteration emulating pointer traversal
"""
def main():
    s = "hello world"
    count = 0
    for _ in s:
        count += 1
    print('Length:', count)

if __name__ == '__main__':
    main()

# Time Complexity: O(n).
# Space Complexity: O(1) additional space.
